moveit_core/robot_model/src/joint_model_group.cpp

```diff
@@ -102,24 +102,9 @@ bool jointPrecedes(const JointModel* a, const JointModel* b)
   return false;
 }
 
-// These are used as filter functions
-const auto is_fixed_joint = [](const auto& jm) { return jm->getVariableCount() == 0; };
-const auto is_mimic_joint = [](const auto& jm) { return jm->getMimic() != nullptr; };
-const auto has_geometry = [](const auto& lm) { return !lm->getShapes().empty(); };
-
-// These are transform functions
-const auto get_name = [](const auto& element) { return element->getName(); };
-const auto get_variable_count = [](const auto& jm) { return jm->getVariableCount(); };
-const auto get_variable_bounds_addr = [](const auto& jm) { return &jm->getVariableBounds(); };
-const auto get_first_variable_index = [](const auto& jm) { return jm->getFirstVariableIndex(); };
-const auto get_variable_names_view = [](const auto& jm) { return views::all(jm->getVariableNames()); };
-const auto get_variable_names_size = [](const auto& jm) { return jm->getVariableNames().size(); };
-
-// Helper to convert a tuple to a pair.  This is useful as a transform after
-// zipping two views inorder to initialize a map.
-const auto tuple_to_pair = [](const auto& element) {
-  return std::make_pair(std::get<0>(element), std::get<1>(element));
-};
+// These is a transform function for calling getName on a Joint or Link Model
+constexpr auto getElementName = [](const auto& element) { return element->getName(); };
```

Comment:
Are we doing snake case or camel case?

Reply:
If it is in the global scope our clang-tidy rules say it has to be camel case.  If it is in the scope of a function it is snake case.  That rule drives me crazy.

Reply:
The reason I want this one to be global scope is it is used by several functions after the compiler.